app-backend/db/src/test/scala/com/azavea/rf/database/CampaignDaoSpec.scala

```diff
@@ -192,64 +196,93 @@ class CampaignDaoSpec
             insertedCampaign <- CampaignDao
               .insertCampaign(
                 campaignCreate.copy(parentCampaignId = None),
-                user
+                parent
               )
             insertedProject <- AnnotationProjectDao
               .insert(
                 annotationProjectCreate.copy(
-                  campaignId = Some(insertedCampaign.id)
+                  campaignId = Some(insertedCampaign.id),
+                  status = AnnotationProjectStatus.Waiting
                 ),
-                user
+                parent
               )
-            campaignCopy <- CampaignDao.copyCampaign(insertedCampaign.id, user)
-            projectCopy <- AnnotationProjectDao.listByCampaign(campaignCopy.id)
+            _ <- AnnotationProjectDao.update(
+              insertedProject.toProject
+                .copy(status = AnnotationProjectStatus.Ready),
+              insertedProject.id
+            )
+            campaignCopies <- children traverse { child =>
+              CampaignDao.copyCampaign(insertedCampaign.id, child)
+            }
+            insertedCampaignAfterCopy <- CampaignDao.unsafeGetCampaignById(
+              insertedCampaign.id
+            )
+            projectCopies <- (campaignCopies traverse { c =>
+              AnnotationProjectDao.listByCampaign(c.id)
+            }) map (_.flatten)
           } yield {
-            (insertedCampaign, insertedProject, campaignCopy, projectCopy)
+            (
+              insertedCampaignAfterCopy,
+              insertedProject,
+              campaignCopies,
+              projectCopies
+            )
           }
 
           val (
             originalCampaign,
             originalProject,
-            copiedCampaign,
-            copiedProject
+            copiedCampaigns,
+            copiedProjects
           ) = copyIO.transact(xa).unsafeRunSync
 
           assert(
-            originalCampaign.name == copiedCampaign.name,
-            "Copy of the campaign worked"
+            originalCampaign.childrenCount == userCreates.size,
+            "Original campaign's children count is the same as the number of users to be added"
           )
-          assert(
-            Set(originalProject.name) == copiedProject.map(_.name).toSet,
-            "Copy of the project worked"
-          )
-          assert(
-            Set(Some(copiedCampaign.id)) == copiedProject
-              .map(_.campaignId)
-              .toSet,
-            "Copy of the project has the id from the copied campaign"
-          )
-          assert(
-            Some(originalCampaign.id) == copiedCampaign.parentCampaignId,
-            "Copy of the campaign has the parent campaign id"
-          )
-          assert(
-            originalCampaign.tags.toSet == copiedCampaign.tags.toSet,
-            "Copy of the campaign has the parent campaign tags"
-          )
-          true
+
+          userCreates.size match {
+            case 0 => true
+            case _ =>
```

Comment:
this test should include an assertion about the children count for the parent campaign, since that will verify the proper function of the trigger

Reply:
@jisantuc, does the assertion above not suffice?

Reply:
... i missed it in the sea of green :man_facepalming: